### PR TITLE
Add hooks to do strict license checking when building on stable branches/releases

### PR DESCRIPTION
**Pull request by sjsharma2001**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -33,11 +33,14 @@ script:
   - |
     if [[ "$TRAVIS_BRANCH" == *"-stable" ]]; then
       export CTLR_VERSION=v$(echo $TRAVIS_BRANCH | sed s/-stable//g)
+      export LICENSE_STRICT="true"
```
Comment (amudukutore) [inline .travis.yml]: I recall we had a discussion on this on an email thread (or perhaps on Slack) as to when this should be enabled. If I recall, we should only enable strict checking on tagged release builds.

File: Makefile [hunk 2]
```diff
@@ -136,6 +136,9 @@ pip_attributions.json: always-build
 		--project-path=$(CURDIR) \
 
 docs/_static/ATTRIBUTIONS.md: flatfile_attributions.json  golang_attributions.json  pip_attributions.json
+        ifeq ($(LICENSE_STRICT), "true")
```
Comment (amudukutore) [inline Makefile]: What happens to this check if LICENSE_STRICT is not defined?

File: .travis.yml (hunk 1 above)
Comment (sjsharma2001) [reply .travis.yml]: I have pushed that change.

**Comment by sjberman** [comment]
@amudukutore I suggested to Sandeep to create his own *-stable branch in his fork, and push the changes to that and ensure the build runs correctly.

**Comment by amudukutore** [comment]
@sjberman - that's a great idea. He can then create a tag on the stable branch on his fork and test the strict=true behavior.

**Comment by sjsharma2001** [comment]
Yes, I will do these tests before merging..

**Comment by sjsharma2001** [comment]
I tested the cases of pushing to "*-stable" branch, as well as creating a tag. Had to make few edits to make it work correctly.

File: Makefile [hunk 3]
```diff
@@ -11,9 +11,23 @@ export BUILD_INFO := $(shell ./build-tools/version-tool build-info)
 
 GO_BUILD_FLAGS=-v -ldflags "-extldflags \"-static\" -X main.version=$(BUILD_VERSION) -X main.buildInfo=$(BUILD_INFO)"
 
-# Allow users to pass in BASE_OS build options (alpine or rhel7)
+# Allow users to pass in BASE_OS build options (alpine or rhel7)
```
Comment (sjberman) [inline Makefile]: Remove the trailing space here.

File: Makefile [hunk 4]
```diff
@@ -135,7 +149,8 @@ pip_attributions.json: always-build
 		--requirements=requirements.txt \
 		--project-path=$(CURDIR) \
 
+
```
Comment (sjberman) [inline Makefile]: Remove this extra line.

File: Makefile [hunk 5]
```diff
@@ -11,9 +11,23 @@ export BUILD_INFO := $(shell ./build-tools/version-tool build-info)
...
 GO_BUILD_FLAGS=-v -ldflags "-extldflags \"-static\" -X main.version=$(BUILD_VERSION) -X main.buildInfo=$(BUILD_INFO)"
 
-# Allow users to pass in BASE_OS build options (alpine or rhel7)
+# Allow users to pass in BASE_OS build options (alpine or rhel7) 
 BASE_OS ?= alpine
 
+# This is for builds not triggered through Travis CI 
+ifndef LICENSE_STRICT
+	LICENSE_STRICT=false
+endif
```
Comment (sjberman) [inline Makefile]: I think there is a simpler way to do this. `LICENSE_STRICT ?= false` should set the variable to false if it isn't already set.

File: Makefile [hunk 6]
```diff
@@ -11,9 +11,23 @@ export BUILD_INFO := $(shell ./build-tools/version-tool build-info)
...
+
+# Flag for level of license approval check 
+# (default if no flag is passed is devel)
+LIC_FLAG=
```
Comment (sjberman) [inline Makefile]: Is this actually doing anything?

Comment (sjsharma2001) [reply Makefile]: I am going to try  with this line removed and see if things work as expected.

File: Makefile (hunk 3 above)
Comment (sjsharma2001) [reply Makefile]: Will do.

File: Makefile (hunk 4 above)
Comment (sjsharma2001) [reply Makefile]: Will do.

File: Makefile (hunk 5 above)
Comment (sjsharma2001) [reply Makefile]: Per our discussion, if testing is taking too long to verify all scenarios, I will defer it for now.

**Comment by sjsharma2001** [comment]
Hi @sjberman I have made all the changes you recommended, and they worked fine in my testing.

**Pull request merged**